### PR TITLE
5.0: Update `django.contrib.auth.base_user`

### PR DESCRIPTION
**Pull request by JaeHyuckSa**
# I have made things!
Update stubs for `django.contrib.auth.base_user` for Django 5.0.

- [x]  `django.contrib.auth.base_user`
  - [x]  `django.contrib.auth.base_user.AbstractBaseUser.get_session_auth_fallback_hash`was added

## Related issues

Refs
- https://github.com/typeddjango/django-stubs/issues/1493

Upstream PR
- https://github.com/django/django/pull/16631

### CODE REVIEW

File: django-stubs/contrib/auth/base_user.pyi [hunk 1]
```diff
@@ -32,6 +32,8 @@ class AbstractBaseUser(models.Model):
     def set_unusable_password(self) -> None: ...
     def has_usable_password(self) -> bool: ...
     def get_session_auth_hash(self) -> str: ...
+    def get_session_auth_fallback_hash(self) -> str: ...
+    def _get_session_auth_hash(self, secret: str | bytes | None = ...) -> str: ...
```
Comment (sobolevn) [inline django-stubs/contrib/auth/base_user.pyi]: We generally don't include protected methods into stubs.

Comment (saJaeHyukc) [reply django-stubs/contrib/auth/base_user.pyi]: I deleted the part that applies! Thank you

**Review by sobolevn — APPROVED** [review]

File: django-stubs/contrib/auth/base_user.pyi [hunk 2]
```diff
@@ -32,6 +32,7 @@ class AbstractBaseUser(models.Model):
     def set_unusable_password(self) -> None: ...
     def has_usable_password(self) -> bool: ...
     def get_session_auth_hash(self) -> str: ...
+    def get_session_auth_fallback_hash(self) -> str: ...
```
Comment (delfick) [inline django-stubs/contrib/auth/base_user.pyi:35]: I believe this should also be `Iterable[str]` https://github.com/django/django/blob/c499184f198df8deb8b5f7282b679babef8384ff/django/contrib/auth/base_user.py#L138-L140

Comment (JaeHyuckSa) [reply django-stubs/contrib/auth/base_user.pyi:35]: https://github.com/typeddjango/django-stubs/pull/2407

Comment (delfick) [reply django-stubs/contrib/auth/base_user.pyi:35]: thanks :)

**Pull request merged**